core/src/main/scala/kafka/api/FetchRequest.scala

```diff
@@ -196,12 +200,14 @@ case class FetchRequest(versionId: Short = FetchRequest.CurrentVersion,
   }
 
   override  def handleError(e: Throwable, requestChannel: RequestChannel, request: RequestChannel.Request): Unit = {
-    val fetchResponsePartitionData = requestInfo.map { case (topicAndPartition, _) =>
-      (topicAndPartition, FetchResponsePartitionData(Errors.forException(e).code, -1, MessageSet.Empty))
+    val responseData = new util.LinkedHashMap[TopicPartition, JFetchResponse.PartitionData]
```

Comment:
Is there an ordering expectation here or something that requires the `LinkedHashMap`? I ask because there's no comment about what is expected here and if I saw the new version of this code I might be tempted to convert it back to a `map` of `requestInfo`.